### PR TITLE
Add basic parsing fuzz target.

### PR DESCRIPTION
**Pull request by frewsxcv**
Run with `cargo fuzz run parse`

https://github.com/php-rust-tools/parser/issues/136

### CODE REVIEW

**Comment by azjezz** [comment]
Thank you for contributing.

This looks okay, but i will have to look more into it later.

Could you please add a GA for the fuzzer?

**Comment by ryangjchandler** [comment]
I'll check this out shortly and give it a run, will need to look into the crate as well.

**Comment by frewsxcv** [comment]
> GA for the fuzzer

@azjezz What is a GA?

**Comment by ryangjchandler** [comment]
> > GA for the fuzzer
> 
> @azjezz What is a GA?

GitHub Action.

**Comment by frewsxcv** [comment]
Can you approve the workflow so it can run?

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -72,3 +72,48 @@ jobs:
...
+              target/
+            key: rust-${{ inputs.rust }}-os-${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: install cargo-fuzz
+        uses: actions-rs/cargo@v1
+        with:
+          command: install
+          args: cargo-fuzz
+
+      - run: cargo fuzz build
```
Comment (azjezz) [inline .github/workflows/tests.yml:119]: we don't need to re-install everything, we can use the same setup as before.

```suggestion
      - name: install cargo-fuzz
        uses: actions-rs/cargo@v1
        with:
          command: install
          args: cargo-fuzz

      - name: fuzz
        uses: actions-rs/cargo@v1
        with:
          command: fuzz
          args: build
```

**Comment by frewsxcv** [comment]
need another approval of the workflow run

**Comment by frewsxcv** [comment]
Updated it so it only runs on nightly, since cargo fuzz only supports nightly

**Comment by frewsxcv** [comment]
It looks like CI on `main` might be broken?

**Pull request closed without merging**